### PR TITLE
Make tab switching multi-platform

### PR DESCRIPTION
**Pull request by Gozala**
- Remove obsolete deck actions.
- Generate diff bindings based on platform.

### CODE REVIEW

File: src/browser/browser.js [hunk 1]
```diff
@@ -94,7 +94,33 @@ define((require, exports, module) => {
...
+  switchTab.toIndex = index => items => switchTab(items, items.get(index));
+  switchTab.toLast = items => switchTab(items, items.last());
+
+
+  let onTabSwitch;
+  {
+    const os = navigator.platform.startsWith('Win') ? 'windows' :
+               navigator.platform.startsWith('Mac') ? 'osx' :
+               navigator.platform.startsWith('Linux') ? 'linux' :
+               '';
```
Comment (fabricedesre) [inline src/browser/browser.js]: we already do that in src/browser/keyboard.js and src/browser/actions.js. Looks like we should move it to a common place.

Comment (Gozala) [inline src/browser/browser.js]: Yeah I was thinking about that too, I'll send a new patch to do so.

File: src/os.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+// Subset of `os` module from node.js and io.js:
+// https://iojs.org/api/os.html
+define((require, exports, module) => {
+  const platform = navigator.platform.startsWith('Win') ? 'win32' :
```
Comment (fabricedesre) [inline src/os.js]: nit: I would not say win32, as it's also valid for 64 bits windows.

File: src/os.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+                   navigator.platform.startsWith('Mac') ? 'darwin' :
+                   navigator.platform.startsWith('Linux') ? 'linux' :
+                   navigator.platform.startsWith('FreeBSD') ? 'freebsd' :
```
Comment (fabricedesre) [inline src/os.js]: ;) did you test that? What about BeOS!?

**Pull request merged**